qiskit/providers/backend_compat.py

```diff
@@ -0,0 +1,279 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Backend abstract interface for providers."""
+
+from __future__ import annotations
+
+from typing import List, Iterable, Any, Dict, Optional
+
+from qiskit.exceptions import QiskitError
+
+from qiskit.providers.backend import BackendV1, BackendV2
+from qiskit.providers.backend import QubitProperties
+from qiskit.utils.units import apply_prefix
+from qiskit.circuit.library.standard_gates import get_standard_gate_name_mapping
+from qiskit.circuit.measure import Measure
+from qiskit.providers.models.backendconfiguration import BackendConfiguration
+from qiskit.providers.models.backendproperties import BackendProperties
+from qiskit.providers.models.pulsedefaults import PulseDefaults
+from qiskit.providers.options import Options
+from qiskit.providers.exceptions import BackendPropertyError
+
+# Some BackendConfiguration objects contain pulse only instructions in their
+# supported_instructions field. Filter these out since they don't go in a
+# TARGET
+PULSE_INSTRUCTIONS = {"acquire", "shiftf", "setf", "play"}
```

Comment:
I also confirmed this but it is surprising that IBM backends report such weird instructions. For example, `shiftf` is not a valid instruction name in the OpenPulse spec. `shiftf` should be `fc`, so I guess this is maybe a special circuit instruction that directly controls `FrameChange` pulse command from circuit payload? Anyways no one can use this instruction neither from circuit nor pulse, as far as I know.

I'm not sure what the `configurations().supported_instructions` represents. This [docstring](https://github.com/Qiskit/qiskit-terra/blob/672e95d089ba0e834915796f2b1d59527f3b4ef2/qiskit/providers/models/backendconfiguration.py#L268) doesn't explain anything, and I don't understand how it differs from `basis_gates`. I think we don't need to check this instructions field.

Reply:
I did this because for ibm backends it was necessary to get the `delay` instruction. I didn't see it listed in the basis gates field for any configuration payload that I looked at but it was in supported instruction. If I don't use it then the test validating delay worked us the converter failed.

Reply:
I'm not sure if it's the best way to do this as I don't really know how the field is used in general. I know for providers/backends I've written its not used, we just used basis gates.

Reply:
I opted to add an explicit flag to add a delay instruction and dropped the use of the `supported_instructions` field in: https://github.com/Qiskit/qiskit-terra/pull/8759/commits/a4db6669a279b6d3d0535ad89c49c0718524c5f0